hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaManager.java

```diff
@@ -322,12 +327,30 @@ public boolean isPartitionReplicaVersionDirty(int partitionId, ServiceNamespace
         return replicaVersions[partitionId].isDirty(namespace);
     }
 
+    @Override
+    public void markPartitionReplicaAsSyncRequired(int partitionId, ServiceNamespace namespace, int replicaIndex) {
+        replicaVersions[partitionId].markAsSyncRequired(namespace, replicaIndex);
+    }
+
     @Override
     // Caution: Returning version array without copying for performance reasons. Callers must not modify this array!
     public long[] getPartitionReplicaVersions(int partitionId, ServiceNamespace namespace) {
         return replicaVersions[partitionId].get(namespace);
     }
 
+    @Override
+    // Caution: Returning version array without copying for performance reasons. Callers must not modify this array!
+    // Mutates the replica version array, so that any replica indexes which require sync are reset to 0
```

Comment:
really nit: I would have parametrized the method above `getPartitionReplicaVersions(int partitionId, ServiceNamespace namespace, boolean resetSync)`